mautrix_instagram/portal.py

```diff
@@ -452,21 +458,31 @@ async def _reupload_instagram_media(self, source: 'u.User', media: RegularMediaI
         return await self._reupload_instagram_file(source, url, msgtype, info, intent)
 
     async def _reupload_instagram_animated(self, source: 'u.User', media: AnimatedMediaItem,
-                                           intent: IntentAPI) -> ReuploadedMediaInfo:
+                                           intent: IntentAPI) -> MediaMessageEventContent:
         url = media.images.fixed_height.webp
         info = ImageInfo(height=int(media.images.fixed_height.height),
                          width=int(media.images.fixed_height.width))
         return await self._reupload_instagram_file(source, url, MessageType.IMAGE, info, intent)
 
     async def _reupload_instagram_voice(self, source: 'u.User', media: VoiceMediaItem,
-                                        intent: IntentAPI) -> ReuploadedMediaInfo:
+                                        intent: IntentAPI) -> MediaMessageEventContent:
         url = media.media.audio.audio_src
         info = AudioInfo(duration=media.media.audio.duration)
-        return await self._reupload_instagram_file(source, url, MessageType.AUDIO, info, intent)
+        waveform = [int(p * 1000) for p in media.media.audio.waveform_data]
+        content = await self._reupload_instagram_file(source, url, MessageType.AUDIO, info, intent)
+        content.extensible_file = ExtensibleFile(
+            file=content.file,
+            name=content.body,
+            mimetype=content.info.mimetype,
+            size=content.info.size,
+        )
+        content.audio = Audio(duration=media.media.audio.duration, waveform=waveform)
+        content.voice = Voice()
```

Comment:
I think for the initial version we should just use custom fields. Adding partial support for extensible events in the dataclasses is a bit messy. The eventual goal is to have classes only for extensible events and then something that automatically converts legacy events to the extensible event classes (so bridges will only have to deal with extensible events).

```suggestion
        content["org.matrix.msc1767.file"] = {
            "url": content.url,
            "name": content.body,
            **(content.file.serialize() if content.file else {}),
            **(content.info.serialize() if content.info else {}),
        }
        content["org.matrix.msc1767.audio"] = {
            "duration": media.media.audio.duration,
            "waveform": waveform,
        }
        content["org.matrix.msc3245.voice"] = {}
```